spring-petclinic/src/main/java/org/springframework/samples/petclinic/service/LoggingAdvice.java

```diff
@@ -0,0 +1,39 @@
+package org.springframework.samples.petclinic.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+public class LoggingAdvice {
+
+	Logger log = LoggerFactory.getLogger(LoggingAdvice.class);
+
+	@Pointcut(value="execution(* org.springframework.samples.petclinic.*.*.*(..))")
+	public void myPointcut() {
+
+	}
+
+	@Around("myPointcut()")
+	public Object applicationLogger(ProceedingJoinPoint pjp) throws Throwable {
+		ObjectMapper mapper = new ObjectMapper();
+
+		String methodName = pjp.getSignature().getName();
+		String className = pjp.getTarget().getClass().toString();
+		Object[] array = pjp.getArgs();
+
+		log.info("method called {} : {}() Arguments : {}", className, methodName, array);
+
+		Object object = pjp.proceed();
+
+		log.info("{} : {}() Response : {}", className, methodName, mapper.writeValueAsString(object));
```

Comment:
We can create a model to keep the attributes organized. You can add a new file called LoggedMethod in the service folder. Method Arguments have a dynamic type so could not serialize into the model. We will need to find a way to format the log information in the next sprints.

```
package org.springframework.samples.petclinic.service;

public class LoggedMethod {
	public String className;
	public String methodName;
	public String response;

	public LoggedMethod(String className, String methodName) {
		this.className = className;
		this.methodName = methodName;
	}

	public void setResponse(String response) {
		this.response = response;
	}
}
```

```suggestion
		LoggedMethod loggedMethod = new LoggedMethod(className, methodName);

		Object object = pjp.proceed();
		String response = mapper.writeValueAsString(object);
		loggedMethod.setResponse(mapper.writeValueAsString(response));

		log.info("{}, \"arguments\": {}", mapper.writeValueAsString(loggedMethod), arguments);
```